src/jdk.incubator.code/share/classes/jdk/incubator/code/internal/CodeModelToAST.java

```diff
@@ -80,48 +82,53 @@ private JCTree invokeOpToJCMethodInvocation(CoreOp.InvokeOp invokeOp) {
         }
         var methodInvocation = treeMaker.App(meth, args.toList());
         if (invokeOp.isVarArgs()) {
-            var lastParam = invokeOp.invokeDescriptor().type().parameterTypes().getLast();
-            if (lastParam instanceof ArrayType varargType) {
-                methodInvocation.varargsElement = typeElementToType(varargType.componentType());
-            } else {
-                Assert.error("Vararg method doesn't have a trailing array type: " + invokeOp.invokeDescriptor());
-            }
+            setVarargs(methodInvocation, invokeOp.invokeDescriptor().type());
+        }
+        if (invokeOp.result().uses().isEmpty()) {
```

Comment:
This part was removed by a previous PR. Now it's going to be reintroduced.

Reply:
good catch - seems like a merge issue